src/system/applications/combat/combat-tracker.ts

```diff
@@ -0,0 +1,156 @@
+import CosmereCombatant from './combatant';
+
+/**
+ * Overrides default tracker template to implement slow/fast buckets and activation button.
+ */
+export default class CosmereCombatTracker extends CombatTracker {
+    // eslint-disable-next-line @typescript-eslint/class-literal-property-style
+    override get template() {
+        return 'systems/cosmere-rpg/templates/combat/combat-tracker.hbs';
+    }
+
+    //modifies data being sent to the combat tracker template to add turn speed, type and activation status and splitting turns between the initiative phases.
+    override async getData(
+        options?: Partial<ApplicationOptions> | undefined,
+    ): Promise<object> {
+        const data = (await super.getData(options)) as {
+            turns: CosmereTurn[];
+            fastPlayers: CosmereTurn[];
+            slowPlayers: CosmereTurn[];
+            fastNPC: CosmereTurn[];
+            slowNPC: CosmereTurn[];
+        };
+        data.turns = data.turns.map((turn) => {
+            const combatant: CosmereCombatant =
+                this.viewed!.getEmbeddedDocument(
+                    'Combatant',
+                    turn.id,
+                    {},
+                ) as CosmereCombatant;
+            const newTurn: CosmereTurn = {
+                ...turn,
+                turnSpeed: combatant.getFlag(
+                    'cosmere-rpg',
+                    'turnSpeed',
+                ) as string,
+                type: combatant.actor.type,
+                activated: combatant.getFlag(
+                    'cosmere-rpg',
+                    'activated',
+                ) as boolean,
+            };
+            //strips active player formatting
+            newTurn.css = '';
+            return newTurn;
+        });
+        data.fastPlayers = data.turns.filter((turn) => {
+            return turn.type == 'character' && turn.turnSpeed == 'fast';
+        });
+        data.slowPlayers = data.turns.filter((turn) => {
+            return turn.type == 'character' && turn.turnSpeed == 'slow';
+        });
+        data.fastNPC = data.turns.filter((turn) => {
+            return turn.type == 'adversary' && turn.turnSpeed == 'fast';
+        });
+        data.slowNPC = data.turns.filter((turn) => {
+            return turn.type == 'adversary' && turn.turnSpeed == 'slow';
+        });
```

Comment:
These should use the `ActorType` and `TurnSpeed` enums for the comparison. Also its generally best practice to use `===` instead of `==`.

Reply:
Fixed enums and replaced all ==